datagovharvester/extract/utils.py

```diff
@@ -0,0 +1,52 @@
+import requests
+import json
+import os
+from datagovharvester.utils.s3_utilities import (
+    upload_dcatus_to_S3,
+    create_s3_client,
+    create_bucket,
+    create_s3_payload,
+)
+
+
+def download_catalog(url):
+    resp = None
+    error = None
```

Comment:
I think having a docstring would be more useful than adding unnecessary initializers for python code.

Reply:
Python is a loosely-typed language.  There is also no concept of variable initialization in Python.  To those points, there are very very very .... very few usecases that you need to explicitly initialize a variable.  Since `resp` and `error` are simple variable assignments, no need to initialize them with `None`.

To track "outputs", you can have a docstring that outlines the inputs/outputs and general purpose of the function... In this case....

```bash
def download_catalog(url):
    ''' Download agency source catalog data
    url (str)     : https link to download data
    resp (str)    : data download
    error (error) : errors encountered when downloading
    '''
    ...the rest of the function...
```

Reply:
... but be more specific, is `error` an instance of error class? or it is a list of errors?  Or something else?

Reply:
Pin this topic to a pairing session for group discussion.